arch/x86/boot/compressed/sl_main.c

```diff
@@ -114,14 +115,53 @@ static void sl_txt_validate_msrs(struct txt_os_mle_data *os_mle_data)
 		sl_txt_reset(SL_ERROR_MSR_INV_MISC_EN);
 }
 
+/*
+ * In order to simplify adding new entries and to minimize the number of
+ * differences between AMD and Intel, the event logs have actually two headers,
+ * both for TPM 1.2 and 2.0.
+ *
+ * For TPM 1.2 this is TCG_PCClientSpecIDEventStruct [1] with Intel's own
+ * TXT-specific header embedded inside its 'vendorInfo' field. The offset to
+ * this field is added to the base address in AMD path, making the code for
+ * adding new events the same for both vendors.
+ *
+ * TPM 2.0 in TXT uses HEAP_EVENT_LOG_POINTER_ELEMENT2_1 structure, which is
+ * normally constructed on the TXT stack [2]. For AMD, this structure is put
+ * inside TCG_EfiSpecIdEvent [3], also in 'vendorInfo' field. The actual offset
+ * to this field depends on number of hash algorithms supported by the event
+ * log.
+ *
+ * [1] https://www.trustedcomputinggroup.org/wp-content/uploads/TCG_PCClientImplementation_1-21_1_00.pdf
+ * [2] http://www.intel.com/content/dam/www/public/us/en/documents/guides/intel-txt-software-development-guide.pdf
+ * [3] https://trustedcomputinggroup.org/wp-content/uploads/TCG_PCClientSpecPlat_TPM_2p0_1p04_pub.pdf
+ */
+#define TCG_PCClientSpecIDEventStruct_SIZE	25
```

Comment:
I have a feeling these macros using number will get flagged in a review and ones that I did in the past did. Might be better to use sizeof(this-and-that) instead.

Reply:
Right. This would be structure defined between pages 108-109 in [1] mentioned above this line. I think the best place to put it would be include/linux/tpm_eventlog.h, I don't think it is already defined anywhere in the tree.

Reply:
Not sure about that, that is rather proprietary format. On the other hand, it still is TPM event log. Maybe the best approach would be to move them all to tpm_eventlog.h and see what maintainers will say about that.

Reply:
+1 (with comments about the TXT one)